ui.frontend/src/view/FormFieldBase.js

```diff
@@ -82,8 +97,39 @@ export default class FormFieldBase extends FormField {
         if (state.value) {
             this._updateValue(state.value);
         }
-        this._updateVisible(state.visible);
-        this._updateEnable(state.enabled);
+        this._updateVisible(state.visible)
+        this._updateEnable(state.visible)
+        this._initializeHelpContent(state);
+    }
+
+    _initializeHelpContent(state) {
+        // Initializing Hint ('?') and long description.
+        this._showHideLongDescriptionDiv(false);
+        if (this.getDescription()) {
+            this._addHelpIconHandler(state);
+        }
+    }
+
+    /**
+     *
+     * @param show If true then <div> containing tooltip(Short Description) will be shown else hidden
+     * @private
+     */
+    _showHideTooltipDiv(show) {
+        this.toggleAttribute(this.getTooltipDiv(), show, 'data-cmp-shortdescription-hidden', true);
+    }
+
+    /**
+     *
+     * @param show If true then <div> containing description(Long Description) will be shown
+     * @private
+     */
+    _showHideLongDescriptionDiv(show) {
+        this.toggleAttribute(this.getDescription(), show, this.longDescriptionHiddenAttribute, true);
+    }
+
+    _isTooltipAlwaysVisible(state) {
+        return state && state.properties && state.properties['afs:layout'] && state.properties['afs:layout'].tooltipVisible;
```

Comment:
this is looking bad. we should not need to go in deep inside object structure using dot notation for extracting basic props.

Reply:
As of now this seems to be the only way to retrieve custom properties.

Reply:
can we create an API `getLayoutProperty()`